src/pytest_mysql/factories/client.py

```diff
@@ -131,8 +131,27 @@ def mysql_fixture(request: FixtureRequest) -> MySQLdb.Connection:
         yield mysql_conn
 
         # clean up after test that forgot to fetch selected data
-        mysql_conn.store_result()
-        mysql_conn.query("DROP DATABASE IF EXISTS %s" % mysql_db)
-        mysql_conn.close()
+        try:
+            if mysql_conn.open:
+                mysql_conn.store_result()
+        except Exception as e:
+            print(str(e))
+        try:
+            if not mysql_conn.open:
+                try:
```

Comment:
this whole block could be just about connecting, not necessarily connecting with dropping and dropping. Now we have three drop database statements :)